playground.sh

```diff
@@ -64,140 +127,277 @@ checkHelm() {
+
+    case "$choice" in
+    k8s)
+      echo "[INFO] Using K8s runtime."
+      runtime="k8s"
+      ;;
+    docker | "") # Empty input defaults to docker
+      echo "[INFO] Using Docker runtime."
+      runtime="docker"
+      ;;
+    *)
+      echo "[ERROR] Invalid choice. Using default: docker."
+      runtime="docker"
+      ;;
+    esac
+    return
+  fi
+
+  # If only Docker is available
+  if [ "${dockerAvailable}" = true ]; then
+    runtime="docker"
+    return
+  fi
+
+  # If only K8s is available
+  if [ "${k8sAvailable}" = true ]; then
+    runtime="k8s"
+    return
+  fi
+}
+
+checkCurrentRuntime() {
+  runtime=""
+
+  # Check if gravitino-playground is running in Docker
+  if command -v docker >/dev/null 2>&1; then
+    if docker compose ls | grep -q "${playgroundRuntimeName}"; then
+      echo "[INFO] gravitino-playground is running in Docker."
+      runtime="docker"
+      return
+    fi
+  fi
+
+  # Check if gravitino-playground is running in K8s
+  if command -v kubectl >/dev/null 2>&1; then
+    if kubectl get namespace "${playgroundRuntimeName}" >/dev/null 2>&1; then
+      if kubectl -n "${playgroundRuntimeName}" get pods | grep -q "Running"; then
+        echo "[INFO] gravitino-playground is running in Kubernetes."
+        runtime="k8s"
+        return
+      fi
+    fi
+  fi
 }
 
 start() {
-  echo "INFO: Starting the playground..."
+  echo "[INFO] Starting the playground..."
+  echo "[INFO] The playground requires ${requiredCpuCores} CPU cores, ${requiredRamGB} GB of RAM, and ${requiredDiskSpaceGB} GB of disk storage to operate efficiently."
 
-  case "$runtime" in
+  checkCurrentRuntime
+  if [ -n "${runtime}" ]; then
+    echo "[ERROR] Playground is already running in ${runtime}. Please stop it first."
+    exit 1
+  fi
+
+  checkRuntime
+  checkPortsInUse
+
+  case "${runtime}" in
   k8s)
     testK8s
     checkHelm
+    checkK8sDisk
+    checkK8sRam
+    checkK8sCpu
     ;;
   docker)
     testDocker
-    checkCompose
-    ports=(8090 9001 3307 19000 19083 60070 13306 15342 18080 18888 19090 13000)
-    for port in "${ports[@]}"; do
-      checkPortInUse ${port}
-    done
+    checkDockerCompose
+    checkDockerDisk
+    checkDockerRam
+    checkDockerCpu
     ;;
   esac
 
-  cd ${playground_dir}
-  echo "Preparing packages..."
+  cd ${playground_dir} || exit 1
+  echo "[INFO] Preparing packages..."
   ./init/spark/spark-dependency.sh
   ./init/gravitino/gravitino-dependency.sh
   ./init/jupyter/jupyter-dependency.sh
 
-  case "$runtime" in
+  case "${runtime}" in
   k8s)
-    helm upgrade --install gravitino-playground ./helm-chart/ \
-      --create-namespace --namespace gravitino-playground \
+    helm upgrade --install ${playgroundRuntimeName} ./helm-chart/ \
+      --create-namespace --namespace ${playgroundRuntimeName} \
       --set projectRoot=$(pwd)
     ;;
   docker)
-    logSuffix=$(date +%Y%m%d%H%m%s)
-    docker-compose up --detach
```

Comment:
`docker-compose` or `docker compose`？
My macOS only have `docker-compose`.
`docker compose` is Linux OS?
Please check it.

Maybe you needs to check operation OS, and add different process.

Reply:
Based on [this documentation](https://docs.docker.com/compose/releases/migrate/), I use `docker-compose`, which works on every operating system.

Reply:
It's not about the OS; it's about the Docker version. Docker version 2 has a built-in `docker compose` command, whereas Docker version 1 does not. For Docker version 1, you need to use `docker-compose`, which is now deprecated.

I added a function to automatically detect the appropriate Docker Compose command.

**Reference**:
[Docker Compose Migration Guide](https://docs.docker.com/compose/releases/migrate/)